app/src/features/charts/UsaChloroplethMap.tsx

```diff
@@ -0,0 +1,228 @@
+      data: [
+        {
+          name: VAR_DATASET,
+          url: props.dataUrl,
+          format: { type: props.dataUrl.split(".").pop() },
+          transform: varTransformer,
+        },
+        {
+          name: GEO_DATASET,
+          transform: geoTransformers,
+          url: "counties-10m.json",
+          format: {
+            type: "topojson",
+            feature: props.stateFips ? "counties" : "states",
+          },
+        },
+        {
+          name: "selected",
+          on: [
+            { trigger: "click", insert: "click" },
+            { trigger: "shiftClick", remove: "true" },
+          ],
+        },
+      ],
+      projections: [
+        {
+          name: "usProjection",
+          type: "albersUsa",
+          fit: { signal: "data('" + GEO_DATASET + "')" },
+          size: {
+            signal:
+              "[" +
+              (width! - LEGEND_WIDTH) +
+              ", " +
+              width! * HEIGHT_WIDTH_RATIO +
+              "]",
+          },
+        },
+      ],
+      scales: [
+        {
+          name: "colorScale",
+          type: "quantize",
+          domain: { data: GEO_DATASET, field: props.varField },
+          range: { scheme: "yellowgreenblue", count: 7 },
+        },
+      ],
+      legends: [legend],
+      marks: [
+        {
+          type: "shape",
+          from: { data: GEO_DATASET },
+          encode: {
+            enter: {
+              tooltip: {
+                signal: tooltipValue,
+              },
+            },
+            update: {
+              fill: [
+                {
+                  test: "indata('selected', 'id', datum.id)",
+                  value: "red",
+                },
+                { scale: "colorScale", field: props.varField },
+              ],
+            },
+            hover: { fill: { value: "pink" } },
+          },
+          transform: [{ type: "geoshape", projection: "usProjection" }],
+        },
+      ],
+      signals: [
+        {
+          name: "click",
+          value: 0,
+          on: [{ events: "*:mousedown", update: "datum" }],
+        },
+        {
+          name: "shiftClick",
+          on: [{ events: "click[event.shiftKey]", update: "datum" }],
+        },
+      ],
+    });
+  }, [
+    width,
+    props.varField,
+    props.legendTitle,
+    props.filterVar,
+    props.filterValue,
+    props.dataUrl,
+    props.operation,
+    props.stateFips,
+    props.numberFormat,
+  ]);
+
+  // TODO: useLayoutEffect or other
+  useEffect(() => {
+    if (myRef && myRef.current) {
+      setWidth(myRef.current.offsetWidth);
+    }
+
+    const handleResize = () => {
+      if (myRef && myRef.current) {
+        setWidth(myRef.current.offsetWidth);
+      }
+    };
+
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, [myRef]);
```

Comment:
I think it should be a pretty quick refactor to extract a hook, do you mind doing that before submitting? (or as a followup)